go-controller/pkg/node/controllers/egressip/egressip.go

```diff
@@ -246,10 +246,19 @@ func (c *Controller) Run(stopCh <-chan struct{}, wg *sync.WaitGroup, threads int
 			return fmt.Errorf("unable to ensure iptables rules for jump rule: %v", err)
 		}
 	}
-	if err := c.RepairNode(); err != nil {
-		// TODO(mk): return error here instead of logging and retry or put in a retry func
-		klog.Errorf("Failed to repair node. Stale configuration maybe present on node: %v", err)
+
+	err = wait.PollUntilContextTimeout(wait.ContextForChannel(stopCh), 1*time.Second, 10*time.Second, true,
```

Comment:
nice! does this mean that we will be able to keep pod from restarting, should this happen and we are able to repair within 10 seconds? 
we are certain that `c.RepairNode()` is idem-potent, right?

Reply:
>does this mean that we will be able to keep pod from restarting, should this happen and we are able to repair within 10 seconds?

it could potentially increase the "start" time on BM envs but I think its worth it to not cause short circuiting due to a race condition for required annotations.

>we are certain that c.RepairNode() is idem-potent, right?
yes, its idem-potent.